### PR TITLE
jUnit tests complite

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/ReturnHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package core.basesyntax;
```
Comment (nacenik) [inline src/test/java/core/basesyntax/ReturnHandlerTest.java:1]: Your test class should be in the same package as the class that it tests. Fix it for all tests 
also. to create a test you can use the power of IDE https://www.jetbrains.com/help/idea/create-tests.html

File: src/test/java/core/basesyntax/ReturnHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    @Test
+    void returnHandle_nullValue_ok() {
+        fruitDB.add("banana", 30);
+        assertThrows(ValidationDataException.class,
+                () -> actionHandler
+                        .executeAction("banana", null));
+    }
+
+    @Test
+    void returnHandle_negativeValue_ok() {
```
Comment (nacenik) [inline src/test/java/core/basesyntax/ReturnHandlerTest.java:54]: negative value - ok? 
ok - means good/success result 
but your result throws an exception.
fix everywhere 
```suggestion
    void returnHandle_negativeValue_notOk() {
```

File: src/test/java/core/basesyntax/ReturnHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    @Test
+    void returnHandle_nullStorage_notOk() {
+        assertThrows(ValidationDataException.class,
+                () -> new BalanceActionHandler(null)
+                        .executeAction("banana", 20));
+    }
+
+    @Test
+    void returnHandle_emptyName_ok() {
+        fruitDB.add("banana", 30);
```
Comment (nacenik) [inline src/test/java/core/basesyntax/ReturnHandlerTest.java:39]: looks like code duplication. You can put this line to the method with `BeforeEach` annotation. 
fix everywhere

File: src/test/java/core/basesyntax/SupplyHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    @Test
+    void supplyHandle_nullStorage_notOk() {
+        assertThrows(ValidationDataException.class,
+                () -> new BalanceActionHandler(null)
+                        .executeAction("banana", 20));
+    }
+
+    @Test
+    void supplyHandle_emptyName_ok() {
+        fruitDB.add("banana", 30);
```
Comment (nacenik) [inline src/test/java/core/basesyntax/SupplyHandlerTest.java:39]: the same about code duplication

File: src/test/java/core/basesyntax/SupplyHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Test
+    void supplyHandle_nullStorage_notOk() {
+        assertThrows(ValidationDataException.class,
+                () -> new BalanceActionHandler(null)
+                        .executeAction("banana", 20));
+    }
+
+    @Test
+    void supplyHandle_emptyName_ok() {
```
Comment (nacenik) [inline src/test/java/core/basesyntax/SupplyHandlerTest.java:38]: the same about method name

File: src/test/java/core/basesyntax/WriteFileTest.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Test
+    void writeFile_emptyData_notOk() {
+        assertThrows(ValidationDataException.class,
+                () -> writeFileServiceTest.writeToFile(new String[0], "path"));
+    }
+
+    @Test
+    void writeFile_nullPath_notOk() {
+        String[] result = {"test", "test"};
```
Comment (nacenik) [inline src/test/java/core/basesyntax/WriteFileTest.java:34]: the same about code duplication

File: src/test/java/core/basesyntax/PurchaseHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    void purchaseHandle_negativeValue_ok() {
+        fruitDB.add("banana", 30);
+        assertThrows(ValidationDataException.class,
+                () -> actionHandler
+                        .executeAction("banana", -4));
+    }
+
+    @Test
+    void purchaseHandle_notContainFruit_ok() {
+        fruitDB.add("apple", 30);
```
Comment (nacenik) [inline src/test/java/core/basesyntax/PurchaseHandlerTest.java:63]: code duplication

File: src/test/java/core/basesyntax/PurchaseHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    @Test
+    void purchaseHandle_negativeValue_ok() {
+        fruitDB.add("banana", 30);
+        assertThrows(ValidationDataException.class,
+                () -> actionHandler
+                        .executeAction("banana", -4));
+    }
+
+    @Test
+    void purchaseHandle_notContainFruit_ok() {
```
Comment (nacenik) [inline src/test/java/core/basesyntax/PurchaseHandlerTest.java:62]: name

File: src/test/java/core/basesyntax/BalanceHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Test
+    void balanceHandler_emptyNameGood_ok() {
+        assertThrows(ValidationDataException.class,
+                () -> actionHandler
+                        .executeAction("", 20));
+    }
+
+    @Test
+    void balanceHandler_nullValueGood_ok() {
```
Comment (nacenik) [inline src/test/java/core/basesyntax/BalanceHandlerTest.java:44]: name

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution, let's improve it

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request closed without merging**